configuration/jsr223.md

```diff
@@ -0,0 +1,210 @@
+
+### Language-Specific Documentation
+
+* [Javascript](jsr223-js.html)
+* [Jython](jsr223-jython.html)
+* Groovy (TODO)
+
+### Script Locations
+
+Scripts should be placed in the 'USERDATA/automation/jsr223' directory. For example, in Linux installations created with a
+package installer, the directory would be `/etc/openhab2/automation/jsr223`.
+
+At system start time the scripts will be loaded in an order based on their name and then top-down through the directory hierarchy. Note that while scripts may be placed in subdirectories of `jsr223`, they should not have the same names as scripts in other directories (this may be changed in the future). For example, with the following directory structure...
+
+```
+automation/jsr223
+  01/
+    script1.py
+  00script.py
+  script2.py
+```
+
+the load order will be: `00script.py`, `01/script1.py`, `script2.py`.
+
+<a name="trigger_types"></a>
+### Trigger Types (all JSR223 languages)
+
+The following trigger types are defined by openHAB (custom triggers can also be defined)
+and take the specified configuration parameters.
+
+|Trigger: `core.ItemCommandTrigger`|
+|-----------------------------|
+|`itemName`| The item name | TEXT |
+|`command` | The command | TEXT |
+
+|Trigger: `core.ItemStateUpdateTrigger`|
+|-----------------------------|
+|`itemName`| The item name | TEXT |
+|`state` | The state | TEXT |
+
+|Trigger: `core.ItemStateChangeTrigger`|
+|-----------------------------|
+|`itemName`| The item name | TEXT |
+|`previousState` | The previous state | TEXT |
+|`state` | The state | TEXT |
+
+|Trigger: `timer.GenericCronTrigger`|
+|-----------------------------|
+|`cronExpression`| The cron expression | TEXT |
+
+|Trigger: `timer.TimeOfDayTrigger`|
+|-----------------------------|
+|`time`| The time in "hh:mm" format | TEXT |
+
+|Trigger: `core.GenericEventTrigger`|
+|-----------------------------|
+|`eventTopic`| default: "smarthome/*" (for channel events: "smarthome/channels/*/triggered") | TEXT |
+|`eventSource`| item name, channel UID | TEXT |
+|`eventTypes`| ItemCommandEvent, ItemStateEvent | TEXT |
+
+### Additional Information
+
+* [Scripted Rule Support](https://github.com/eclipse/smarthome/wiki/Scripted-Rule-Support) - 
+    Early documentation on JSR223 usage in openHAB 2. Includes IDE setup instructions.
+
+<a name="presets"></a>
+### Predefined Script Variables (all JSR223 languages)
+
+To faciliate JSR223 scripting, several openHAB-related variables are automatically predefined. These presets include:
+* [Default](#default_presets)
+* [`RuleSimple`](#rulesimple_presets)
+* [`RuleSupport`](#rulesupport_presets)
+* [`RuleFactories`](#rulefactories_presets)
+
+<a name="default_presets"></a>
+#### Default Variables (no preset loading required)
+
+Variable | Description |
+---------|-------------|
+`State` | `org.eclipse.smarthome.core.types.State` |
+`Command` | `org.eclipse.smarthome.core.types.State` |
+`DateTime` | `org.joda.time.DateTime` (if Jodatime is available) |
```

Comment:
There is conditional logic in the source code for the available of the Joda library. I don't know if this was because of potential OSGI classloader issues for some other reason. Maybe @smerschjohann can comment.